go/vt/vtctl/reparentutil/replication.go

```diff
@@ -150,7 +150,12 @@ func ReplicaWasRunning(stopStatus *replicationdatapb.StopReplicationStatus) (boo
 		return false, vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "could not determine Before state of StopReplicationStatus %v", stopStatus)
 	}
 
-	return stopStatus.Before.IoState == int32(mysql.ReplicationStateRunning) || stopStatus.Before.SqlState == int32(mysql.ReplicationStateRunning), nil
+	// v13 and older tablets will only provide the binary IoThreadRunning and
+	// SqlThreadRunning values while v14 and newer tablets will provide the
+	// non-binary replication states.
+	// This backwards compatible check can be removed in v15+.
+	return ((stopStatus.Before.IoState == int32(mysql.ReplicationStateRunning) || stopStatus.Before.IoThreadRunning) ||
+		(stopStatus.Before.SqlState == int32(mysql.ReplicationStateRunning) || stopStatus.Before.SqlThreadRunning)), nil
```

Comment:
We can get rid of this here, if we first converted the Before value using `ProtoToReplicationStatus` and just used the IoState. That will ensure that we have the upgrade downgrade logic, all in one file. WDYT? I am going ahead with the change, we can revert if you don't agree